test/zero-downtime-flagd-proxy/zd_test.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/sh
+
+set -eu
+
+# Store the flagd-proxy image to a helper variable
+FLAGD_PROXY_IMG_ORIGINAL=$FLAGD_PROXY_IMG
+
+# Deploy flagd-proxy and the client pod
+kubectl delete ns $ZD_TEST_NAMESPACE_FLAGD_PROXY --ignore-not-found=true
+kubectl create ns $ZD_TEST_NAMESPACE_FLAGD_PROXY
+envsubst < test/zero-downtime-flagd-proxy/manifests/proxy/deployment.yaml | kubectl apply -f - -n $ZD_TEST_NAMESPACE_FLAGD_PROXY
+kubectl apply -f test/zero-downtime-flagd-proxy/manifests/proxy/service.yaml -n $ZD_TEST_NAMESPACE_FLAGD_PROXY
+kubectl apply -f test/zero-downtime-flagd-proxy/manifests/proxy/flag-config.yaml -n $ZD_TEST_NAMESPACE_FLAGD_PROXY
+kubectl wait --for=condition=available deployment/flagd-proxy -n $ZD_TEST_NAMESPACE_FLAGD_PROXY --timeout=60s
+envsubst < test/zero-downtime-flagd-proxy/manifests/pod.yaml | kubectl apply -f - -n $ZD_TEST_NAMESPACE_FLAGD_PROXY
+kubectl wait --for=condition=ready pod/zd-test -n $ZD_TEST_NAMESPACE_FLAGD_PROXY --timeout=30s
+
+# Wait until connections from client to flagd-proxy are established
+sleep 20
```

Comment:
is that sleep still required, as we wait for the `zd-test` pod to be ready in the previous command?

Reply:
It's safe to have it here, as the client still needs to connect to the flagd-proxy after it is ready. To be sure we do not update the Deployment of flagd-proxy before the client is connected, we should wait some time here

Reply:
Ok I was just thinking that this value might be overly cautious - maybe as a follow up, and when that test is executed regularly in the pipeline we will be able to reduce it